### PR TITLE
Fixes/Cleans up the Docker file

### PR DESCRIPTION
**Pull request by PrahlM93**
This PR does the following:
* Fixes the Docker file so that it can be run on a Docker host and accessible externally
* Adds a host option to runserver.py (required for the port forwarding to work in Docker)
* Turns off app debug by default as this should not be run this way with Docker
* Copies the source code from the working directory instead of `git clone` in order to allow the developer to test local changes in Docker before committing code

Eventually, this should run behind Apache + mod_wsgi, but this at least gets Docker in a functioning state.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,14 @@
 FROM fedora
 # cffi>=1.1.0->cryptography->fedmsg->-r
-RUN dnf install -y gcc python-devel libffi-devel openssl-devel git gcc-c++ redhat-rpm-config
+RUN dnf install -y gcc python-devel libffi-devel openssl-devel gcc-c++ redhat-rpm-config && \
+    dnf autoremove -y && \
+    dnf clean all -y
```
Comment (pypingou) [inline Dockerfile]: Why these two lines? I'm not sure I see the added value

File: runserver.py [hunk 2]
```diff
@@ -46,5 +50,5 @@
 
 
 from anitya.app import APP
-APP.debug = True
```
Comment (pypingou) [inline runserver.py]: I'm actually :-1: to this, runserver is meant for dev not for actually running the app, so debugging here is pretty useful :)

**Comment by pypingou** [comment]
Thanks for working on this, I do want to point out though that so far the docker image was provided as a way for people to quickly and easily get anitya running to work on it.
It was not and is not meant to be use in prod, that's why it uses runserver which is definitely not meant for prod and why debugging is enabled.

If you want to run the application in prod, with or without docker, you should really look into wsgi or something like gunicorn.

File: Dockerfile (hunk 1 above)
Comment (PrahlM93) [inline Dockerfile]: This in theory is to make the container smaller by removing cached files.

**Comment by pypingou** [comment]
Okido, thanks for these changes :)

**Comment by PrahlM93** [comment]
@pypingou, thanks for the feedback. I'll probably work on getting mod_wsgi up in the container some time in the future.

**Comment by pypingou** [comment]
@PrahlM93 would there be a way to have two docker files then? One for prod and the other (I guess the actual one) for dev?

**Comment by PrahlM93** [comment]
@pypingou, I think so, but I'm thinking Vagrant would be more appropriate to set up a dev environment. Perhaps Docker becomes production, and Vagrant becomes the way to dev?

**Pull request merged**